src/input_parsers/OnnxParser.cpp

```diff
@@ -0,0 +1,1265 @@
+                {
+                    dimSize = oldShape[i];
+                }
+            }
+        }
+        newShape.append(dimSize);
+    }
+
+    if (inferredIndex != -1)
+    {
+        newShape[inferredIndex] = tensorSize(oldShape) / tensorSize(newShape);
+    }
+    return newShape;
+}
+
+void checkEndianness()
+{
+    int num = 1;
+    bool systemIsLittleEndian = *(char *)&num == 1;
+    if (! systemIsLittleEndian)
+    {
+        String errorMessage = "Support for Onnx files on non-little endian systems is not currently implemented on the command line";
+        throw MarabouError(MarabouError::ONNX_PARSE_ERROR, errorMessage.ascii());
+    }
+}
+
+const onnx::AttributeProto* findAttribute(onnx::NodeProto& node, String name, onnx::AttributeProto_AttributeType expectedType)
+{
+    for (const onnx::AttributeProto& attr : node.attribute())
+    {
+        if (attr.name() == name.ascii())
+        {
+            if (attr.type() != expectedType)
+            {
+                illTypedAttributeError(node, attr, expectedType);
+            }
+            return &attr;
+        }
+    }
+    return nullptr;
+}
+
+float getFloatAttribute(onnx::NodeProto& node, String name, float defaultValue)
+{
+    const onnx::AttributeProto* attr = findAttribute(node, name, onnx::AttributeProto_AttributeType_FLOAT);
+    if (attr == nullptr)
+    {
+        return defaultValue;
+    }
+    return (*attr).f();
+}
+
+int getIntAttribute(onnx::NodeProto& node, String name, int defaultValue)
+{
+    const onnx::AttributeProto* attr = findAttribute(node, name, onnx::AttributeProto_AttributeType_INT);
+    if (attr == nullptr)
+    {
+        return defaultValue;
+    }
+    return (*attr).f();
+}
+
+const onnx::TensorProto& getTensorAttribute(onnx::NodeProto& node, String name)
+{
+    const onnx::AttributeProto* attr = findAttribute(node, name, onnx::AttributeProto_AttributeType_INT);
+    if (attr == nullptr)
+    {
+        missingAttributeError(node, name);
+    }
+    return (*attr).t();
+}
+
+Vector<int> getIntsAttribute(onnx::NodeProto& node, String name, Vector<int>& defaultValue)
+{
+    const onnx::AttributeProto* attr = findAttribute(node, name, onnx::AttributeProto_AttributeType_INT);
+    if (attr == nullptr)
+    {
+        return defaultValue;
+    }
+
+    Vector<int> result;
+    for (int i = 0; i < (*attr).ints_size(); i++)
+    {
+        result.append((*attr).ints(i));
+    }
+    return result;
+ }
+
+/*******************
+ * Private methods *
+ *******************/
+
+/**
+ * @brief Initialises the mapping from node names to nodes and
+ * checks the invariant that there are no duplicate names.
+ */
+void OnnxParser::initialiseMaps()
+{
+    Set<std::string> names;
+    for (const onnx::NodeProto& node : _network.node())
+    {
+        const std::string name = node.name();
+        if (names.exists(name))
+        {
+            String errorMessage = "nodes in Onnx network must have a unique name";
```

Comment:
Capitalize nodes in the error message. Also, below.

Reply:
Fixed in both places.